docs/source/developers/bug_reports.rst

```diff
@@ -186,10 +176,109 @@ Good reproducible examples or minimal bug reports can be found in next tabs:
 
 * `Python: Craft Minimal Bug Reports by Matthew Rocklin <https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports>`_
 * `R: Tidyverse: Make a reprex <https://www.tidyverse.org/help/#reprex>`_
 * `R: Tidyverse's Reprex do's and don'ts <https://reprex.tidyverse.org/articles/reprex-dos-and-donts.html>`_
 * `Mozilla's bug-reporting guidelines <https://developer.mozilla.org/en-US/docs/Mozilla/QA/Bug_writing_guidelines>`_
+
+.. _identify-component:
+
+Identify Arrow component
+++++++++++++++++++++++++
+
+Arrow is an expansive project supporting many languages and organized into a
+number of components. Identifying the affected component(s) helps new issues
+get attention from appropriate contributors.
+
+* Use the **Component** field to indicate the area of the project that your
+  issue pertains to (for example "Python" or "C++").
+* Also prefix the issue title with the component name in brackets, for example
+  ``[Python] issue summary`` ; this helps when navigating lists of open issues,
+  and it also makes our changelogs more readable. Most prefixes are exactly the
+  same as the **Component** name, with the following exceptions:
+
+  * **Component:** Continuous Integration — **Summary prefix:** [CI]
+  * **Component:** Developer Tools — **Summary prefix:** [Dev]
+  * **Component:** Documentation — **Summary prefix:** [Docs]
+
+.. _affected-version:
+
+Identify affected version
++++++++++++++++++++++++++
+
+If you're reporting something that used to work in a previous version
+but doesn't work in the current release, you can add the **Affects version**
+field to identify the earliest known version where the bug is observed.
+For feature requests and other proposals, leave **Affects version** empty as
+it is not applicable.
+
+.. _issue-lifecycle:
+
+Issue lifecycle
+===============
+
+Both bug reports and feature requests follow a defined lifecycle. The issue
+**Status** field is used to document the current state of the issue, while the
+**Resolution** field indicates the outcome of issues that have reached
+terminal status. 
+
+
+.. _issue-status:
+
+Issue Status
+++++++++++++
+
+The Arrow project uses the following statuses in Jira to indicate what has -
+and will be - done on an issue:
+
+* **Open** - This is the initial issue state, prior to a contributor assigning
+  the issue and starting progress. Issues in this state should be unassigned.
+* **In progress** - At the time a contributor self-assigns an issue, the status
+  should be set to In progress by clicking the **Start progress** button. All
+  issues in this status should have an assignee - unassigned issues will be
+  set back to a status of Open. Issues remain "in progress" until resolved or
+  closed, including during review of pull requests.
+* **Resolved** - This is a terminal status indicating action has been taken
+  on the issue, which is now considered completed. Issues in a resolved status
+  may have the following resolution codes set:
```

Comment:
Is that true for non-defect issues (e.g., feature requests)? If a status of "Resolved" should map to a resolution of "Fixed" 100% of the time, that's easy enough to automate (in addition to document).

Reply:
Yes, it is. The merge script does that. Besides, I don't think "Implemented" or "Done" would convey any useful subtlety compared to "Fixed".

Reply:
Hmm, I'm not entirely sure what to make of [ARROW-16790](https://issues.apache.org/jira/browse/ARROW-16790) in this context. The PR was merged, but the resolution is "Resolved" instead of "Fixed".

I've updated this page to direct usage of "Fixed" when resolving an issue. I scanned [this list](https://issues.apache.org/jira/issues/?jql=project%20%3D%20ARROW%20and%20resolution%20!%3D%20Fixed%20and%20status%20%3D%20Resolved%20order%20by%20resolved%20desc) for examples where that's not set, and it seems to mostly be manually-set resolutions, either using a status of "Resolved" when it should be more properly be "Closed" (e.g., "Information Provided" or "Won't Fix" resolutions), or because the issue was resolved by work done on a different issue (and the PR referenced the other issue, not the one with the incorrect resolution).

Reply:
[ARROW-16790](https://issues.apache.org/jira/browse/ARROW-16790) should ideally be marked "Fixed". But it seems that would require reopening the issue... let's not bother with this one perhaps.